### PR TITLE
HDXDSYS-1086 Add HNO freeform column to hapi-pipelines

### PR DESCRIPTION
**Pull request by mcarans**
Also includes populating provider_admin1_name and provider_admin2_name

Note that I've used a cut down test file with MLI, SDN and UKR because the full file is too big to be used for a test

### CODE REVIEW

File: src/hapi/pipelines/database/humanitarian_needs.py [hunk 1]
```diff
@@ -38,21 +41,40 @@ def __init__(
...
+                admin_level = "admintwo"
+                admin_code = row["Admin 2 PCode"]
+            case _:
+                return None
+        admin2_ref = self._admins.get_admin2_ref(
             admin_level, admin_code, dataset_name, errors
         )
+        if admin2_ref is None:
+            if admin_level == "adminone":
+                admin_code = f"{countryiso3}-XXX"
```
Comment (turnerm) [inline src/hapi/pipelines/database/humanitarian_needs.py:69]: You should use the [method](https://github.com/OCHA-DAP/hapi-pipelines/blob/main/src/hapi/pipelines/database/admins.py#L190)

File: tests/test_main.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def test_humanitarian_needs(self, configuration, folder, pipelines):
         count = session.scalar(
             select(func.count(DBHumanitarianNeeds.resource_hdx_id))
         )
-        check.equal(count, 139085)
+        check.equal(count, 44938)
```
Comment (turnerm) [inline tests/test_main.py:188]: Maybe add a comment that the test file only has 3 countries

**Review by turnerm — APPROVED** [review]

File: src/hapi/pipelines/database/humanitarian_needs.py [hunk 3]
```diff
@@ -63,44 +85,37 @@ def populate(self) -> None:
...
-            population_group = row["Population Group"]
-            if population_group == "ALL":
-                population_group = "all"
+            admin2_ref = self.get_admin2_ref(row, dataset_name, errors)
+            provider_admin1_name = row["Admin 1 Name"]
+            if provider_admin1_name is None:
+                provider_admin1_name = ""
+            provider_admin2_name = row["Admin 2 Name"]
+            if provider_admin2_name is None:
+                provider_admin2_name = ""
```
Comment (b-j-mills) [inline src/hapi/pipelines/database/humanitarian_needs.py:108]: You can use the [get_provider_name](https://github.com/OCHA-DAP/hapi-pipelines/blob/a37ffa1a791e6957f36d625d8a9080e2f9389298/src/hapi/pipelines/utilities/provider_admin_names.py#L4) function for this

**Review by b-j-mills — APPROVED** [review]

File: src/hapi/pipelines/database/humanitarian_needs.py (hunk 1 above)
Comment (mcarans) [reply src/hapi/pipelines/database/humanitarian_needs.py:69]: Good idea. I've also made one for admin2 to location.

File: tests/test_main.py (hunk 2 above)
Comment (mcarans) [reply tests/test_main.py:188]: Good idea!

File: src/hapi/pipelines/database/humanitarian_needs.py (hunk 3 above)
Comment (mcarans) [reply src/hapi/pipelines/database/humanitarian_needs.py:108]: Good idea!

**Pull request merged**